### PR TITLE
expression: support pushdown repeat to tiflash

### LINKED ISSUE #36023: Support pushdown repeat to tiflash
## Feature Request
related to https://github.com/pingcap/tiflash/issues/5114

### PR DESCRIPTION
**Pull request by likzn**
### What problem does this PR solve?

Issue Number: close #36023

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
expression: support pushdown repeat to tiflash
```

### CODE REVIEW

File: expression/expression.go [hunk 1]
```diff
@@ -1055,7 +1055,7 @@ func scalarExprSupportedByFlash(function *ScalarFunction) bool {
 
 		ast.Sqrt, ast.Log, ast.Log2, ast.Log10, ast.Ln, ast.Exp, ast.Pow, ast.Sign,
 		ast.Radians, ast.Degrees, ast.Conv, ast.CRC32,
-		ast.JSONLength,
+		ast.JSONLength, ast.Repeat,
```
Comment (guo-shaoge) [inline expression/expression.go:1058]: Please also add an integration test. Like this: https://github.com/pingcap/tidb/pull/35872/files#diff-fa42dbc5f88d5aef4ee1cb7452ee4497fc545324b76bee0e44f48e5f1af9f512R2729.

Comment (likzn) [reply expression/expression.go:1058]: Done

**Review by SeaRise — APPROVED** [review]

**Review by guo-shaoge — APPROVED** [review]

**Review by SeaRise — APPROVED** [review]

**Comment by guo-shaoge** [comment]
@likzn Looks like we have conflicts, please resove it. And I will merge it as soon as possible.

**Review by SeaRise — APPROVED** [review]

**Comment by likzn** [comment]
> @likzn Looks like we have conflicts, please resove it. And I will merge it as soon as possible.

Done.

**Review by windtalker — APPROVED** [review]
LGTM

**Review by SeaRise — APPROVED** [review]

**Review by SeaRise — APPROVED** [review]

**Pull request merged**